### ISSUE TITLE
Pre-flight failure when using LongPolling transport

### ISSUE THREAD

**Issue by gs2810** [cometd/cometd#704]
Hello,

I'm getting the following error when trying to initiate the handshake over long-polling with the JavaScript client.

> XMLHttpRequest cannot load http://10.4.4.4/api/handshake. Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://localhost:8080' is therefore not allowed access.

The server is currently set-up to only accept long-polling, and the "callback-polling" fallback eventually returns "403:Handshake denied".

I've also tried to set up my own transport to configure the long-polling issued HTTP headers, but ran into the same issue.

There doesn't seem to be any information in the Docs relating to the issue, nor any similar issues reported here and on StackOverflow.com. Any idea what might be causing this?

Let me know if you need any additional information.

Thanks

**Comment by sbordet** [comment]
Have you read [this](https://docs.cometd.org/current/reference/#_javascript_transports_cross_origin) and configured `CrossOriginFilter` ?

**Comment by gs2810** [comment]
Hi Simone,

Thank you for such a swift response.
Seems you were right that the issue was being caused by CORS.

We've contacted the server team and requested them to configure the server accordingly; in the meantime, we were able to workaround the issue using Fiddler, as per [this blog post](http://simplyaprogrammer.com/2013/12/using-fiddler-to-force-web-service-to.html).

I'll close the issue for now and re-open if the server configuration doesn't solve the issue.
